releasenotes/notes/fix_unique_circuit_naming-3072ba931bd30c43.yaml

```diff
@@ -0,0 +1,15 @@
+---
+fixes:
+  - |
+    Fixes issue where binding to circuits in 'execute' would cause all
+    the bound circuits to get the same circuit name, which meant the
+    result names were also not unique. This fix causes the circuit to
+    be named like:
+
+       <base name>-<class instance no.>[-<pid name>]
+
+    where <base name> is the name supplied by the "name" keyword,
+    otherwise it defaults to "circuit". The class instance number gets
+    incremented every time an instance of the class is generated. <pid
+    name> is appended if called outside the main process in non win32
+    systems.
```

Comment:
Can you also add an upgrade note that `bind_` and `assign_parameters` will not update the names of the output circuit?

Reply:
This actually updated the name if inplace was false, but maybe it should just occur in qobj assembly?

Reply:
That was a typo on my part. Updating names for on bind/assign (if `inplace=False` ) seems like the right behavior, I had meant to ask for the upgrade note so users of `bind_`/`assign_` would be aware of the change.

Reply:
ok, sounds good